config/checkstyle-non-main-files-suppressions.xml

```diff
@@ -111,6 +111,12 @@
   <!-- We will preserve formatting of archived release notes -->
   <suppress id="lineLength" files="[\\/]releasenotes_old_8\-0_8\-34\.xml"/>
 
+  <!-- All xdoc checks/filters files are generated from templates and have escaped characters that
+       make the lines long. We still validate index.xml and template files.
+       until https://github.com/checkstyle/checkstyle/issues/13426 -->
```

Comment:
Sorry, not a good reason.
There might be some links, but we should advanced nearby suppressions.

Let's share here some examples of violations

Reply:
it's not links that are the problem, it's code blocks. I pushed new commit for you to see
```
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:30: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:151: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:173: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:181: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:211: Line should not be longer than 100 symbols [lineLengthXml]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:233: Line should not be longer than 100 symbols [lineLengthXml]
[INFO]      [echo] Checkstyle finished (checkstyle-checks.xml) : 03/08/2023 01:39:11 PM
[INFO]      [echo] Checkstyle started (checkstyle-non-main-files-checks.xml): 03/08/2023 01:38:38 PM
[INFO] [checkstyle] Running Checkstyle  on 3966 files
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:30: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:151: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:173: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:181: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:211: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs/checks/annotation/annotationusestyle.xml:233: Line should not be longer than 100 symbols [lineLength]

```

Reply:
I don't see reason for suppression, we change such lines, and we should simply wrap them to new lines.

Only urls are ok to violate this rule, as we don't know how to wrap them.